.buildkite/build-static-release.sh

```diff
@@ -28,7 +32,23 @@ esac
 
 echo will run with $CONFIG_OPTS
 
-./bazel build //main:sorbet --strip=always $CONFIG_OPTS
+case "$platform" in
+  darwin-x86_64|darwin-arm64)
+    TARGET_PLATFORM=darwin-x86_64 ./bazel build //main:sorbet --strip=always $CONFIG_OPTS
+    mv bazel-bin/main/sorbet sorbet_x86_64
+
+    ./bazel clean --expunge
+
+    TARGET_PLATFORM=darwin-arm64 ./bazel build //main:sorbet --strip=always $CONFIG_OPTS
+    mv bazel-bin/main/sorbet sorbet_arm64
+
+    lipo -create -output bazel-bin/main/sorbet sorbet_x86_64 sorbet_arm64
+    rm sorbet_x86_64 sorbet_arm64
```

Comment:
`bazel-bin/` is Bazel's domain, we're not permitted to move files into or out of it.  I realize that here we're not executing Bazel again, so it shouldn't matter, but I think we should respect Bazel's invariants here.

So the darwin branch here should `cp` files out of the sandbox and use some made-up-file for `lipo -create -output`, and then communicate that to the bit that copies into `gems/sorbet-static/libexec`, below.  (Some changes may be required for the Linux branch below as well.)